api/v1alpha3/types.go

```diff
@@ -459,6 +459,13 @@ type AvailabilityZone struct {
 // VMIdentity defines the identity of the virtual machine, if configured.
 type VMIdentity string
```

Comment:
Should this have kubebuilder markers for enum and default? Something like the following:

```go
// +kubebuilder:validation:Enum=None;SystemAssigned
type VMIdentity string

// +kubebuilder:default=None
Identity VMIdentity `json:"identity,omitempty"`
```

Reply:
that's a good idea

Reply:
we have generally been using webhooks for setting defaults, not using this annotation, wdyt?

Reply:
Defaulting in the webhooks makes sense to me. Mosts important thing to me was the enum for code gen and documentation.

Reply:
is this resolved?

Reply:
I added the default as suggested